sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -993,6 +994,14 @@ class CodegenContext {
     }
   }
 
+  /**
+   * Returns the code for subexpression elimination after splitting it if necessary.
+   */
+  def subexprFunctionsCode: String = {
+    // Wholestage codegen does not allow subexpression elimination
```

Comment:
Could you elaborate on your thought on this comment?   
IMHO, this change is not directly related to whether wholestage codegen performs subexpression elimination by controlling in other places.

Reply:
The reason of this comment is that here we are passing the `InternalRow` to the variables and not the `currentVars` which would be needed in whole-stage codegen. This is done because in subexpression elimination we always put an `InternalRow` as argument. For this reason, whole-stage codegen disables subexpression elimination.
So I put here this comment in order to highlight that if in the future there will be a work to support subexpression elimination also in wholestage codegen, we need to modify this method too.